src/core_modules/capture-core/metaData/RenderFoundation/CustomForm.js

```diff
@@ -34,13 +34,17 @@ export class CustomForm {
      *
      * @memberof CustomForm
      */
-    set data(html: string) {
-        const data = parseHtml(html, {
-            onTransform: this.transformNode,
+    setData(html: string, transformFunction: () => ?React.Element<'FormField'>) {
+        this._data = parseHtml(html, {
+            onTransform: transformFunction,
             allowScript: true,
         });
+    }
+
+    set data(data: Object) {
```

Comment:
Can you help me understand why `data` is not of type `Data` here? In both the initialisation and the getter type `Data` is used

Reply:
Sorry, this one slipped through my filters. The whole set-method is removed as we already have a function to set the data. Thanks for the feedback!